frontend/public/components/monitoring/alerting.tsx

```diff
@@ -1338,6 +1361,101 @@ const AlertsPage_: React.FC<Alerts> = ({ data, loaded, loadError }) => {
     },
   ];
 
+  const severityRowFilter: RowFilter = {
+    filterGroupName: t('public~Severity'),
+    items: [
+      { id: AlertSeverity.Critical, title: t('public~Critical') },
+      { id: AlertSeverity.Warning, title: t('public~Warning') },
+      { id: AlertSeverity.Info, title: t('public~Info') },
+      { id: AlertSeverity.None, title: t('public~None') },
+    ],
+    reducer: ({ labels }: Alert | Rule) => labels?.severity,
+    type: 'alert-severity',
+  };
+
+  const alertsRowFilters: RowFilter[] = [
+    {
+      defaultSelected: [AlertStates.Firing],
+      filterGroupName: t('public~Alert State'),
+      items: [
+        { id: AlertStates.Firing, title: t('public~Firing') },
+        { id: AlertStates.Pending, title: t('public~Pending') },
+        { id: AlertStates.Silenced, title: t('public~Silenced') },
+      ],
+      reducer: alertState,
+      type: 'alert-state',
+    },
+    severityRowFilter,
+    {
+      defaultSelected: [AlertSource.Platform],
+      filterGroupName: t('public~Source'),
+      items: [
+        { id: AlertSource.Platform, title: t('public~Platform') },
+        { id: AlertSource.User, title: t('public~User') },
+      ],
+      reducer: alertSource,
+      type: 'alert-source',
+    },
+  ];
+
+  const AlertTableRow: RowFunction<Alert> = ({ index, key, obj, style }) => {
```

Comment:
Prefer to keep this as a separate component to keep `AlertsPage` from growing too large.